plugins/star-rating/frontend/public/javascripts/countly.models.js

```diff
@@ -87,6 +89,162 @@
         });
     };
 
+    /**
+     * This is for fetching feedback comments objects from server side 
+     * @namespace starRatingPlugin
+     * @method requestFeedbackData
+     * @param {}
+     * @return {func} ajax func to request data and store in _fd
+     */
+    starRatingPlugin.requestFeedbackData = function() {
+        var periodString = countlyCommon.getPeriodForAjax();
+        
+        // returning promise
+        return $.ajax({
+            type: "GET",
+            url: countlyCommon.API_URL + "/o/web-feedback/data",
+            data: {
+                api_key: countlyGlobal['member'].api_key,
+                app_key: countlyGlobal['apps'][store.get("countly_active_app")].key,
+                app_id: countlyCommon.ACTIVE_APP_ID
+            },
+            success: function (json) {
+                _fd = json;
+            }
+        })
+    }
+
+    /**
+     * This is for fetching feedback comments objects from server side 
+     * @namespace starRatingPlugin
+     * @method requestFeedbackData
+     * @param {}
+     * @return {func} ajax func to request data and store in _fd
+     */
+    starRatingPlugin.requestSingleWidget = function(id, callback) {
+        // returning promise
+        return $.ajax({
+            type: "GET",
+            url: countlyCommon.API_URL + "/o/web-feedback/widget",
+            data: {
+                widget_id: id,
+                app_id: countlyCommon.ACTIVE_APP_ID,
+                app_key: countlyGlobal['apps'][store.get("countly_active_app")].key,
+                api_key: countlyGlobal['member'].api_key,
+            },
+            success: function (json) {
+                callback(json);
+            }
+        })
+    }
+
+    starRatingPlugin.createFeedbackWidget = function(feedbackWidget, callback) {
+        return $.ajax({
+            type: "GET",
+            url: countlyCommon.API_URL + "/i/web-feedback/widgets/create",
+            data: {
+                api_key: countlyGlobal['member'].api_key,
+                app_key: countlyGlobal['apps'][store.get("countly_active_app")].key,
```

Comment:
no need to pass app_key when not needed, let's not expose it that much, same in all other places